### DISCUSSION TITLE
Relays and ioBroker

### DISCUSSION THREAD

**Discussion by DieterB61** [raomin/ESPAltherma#326] [Q&A]
Hi,

after successfully logging my data using an ESP32 board, I ordered a board with 4 relays to realize power control using the X85 connector. Unfortunately I got a board with ESP-12F (ESP8266). As I got my money back without having to return the board I gave it a try to use it for ESPAltherma.

To see if everything is working, I first flashed Tasmota and could set all 4 relays from ioBroker using the Sonoff-MQTT-Adapter. The ioBroker MQTT Adapter did not give me all possibilities! For configuration see https://templates.blakadder.com/ESP12F_Relay_X4.html

Then I flashed ESPAltherma using the nodemcuv2 environement.
In a first attempt I tried to set the relais using the SG Pins. I set:
#define PIN_SG1 16// Pin connected to dry contact SG 1 relay (normally open) 
#define PIN_SG2 14// Pin connected to dry contact SG 2 relay (normally open) 
Pin 14 and 16 worked for Tasmota. For RX and TX Pins 4 and 5 where used.

The logging works in this configuration, but I'm not able to change the status of the relays when I change the SG -> Set Value within ioBroker. I've tried SG_RELAY_HIGH_TRIGGER and SG_RELAY_LOW_TRIGGER.

Any idea what I'm doing wrong?

**Comment by DieterB61** [comment]
Additional question: has the board to be connected to X10A to test the switching of the SG relais?

Today I tried to add [javimurcia](https://github.com/javimurcia) X85 extension but I got the same behavior. Relais did nothing when changing ths x85a -> set value.
Here one picture how it looks like in ioBroker MQTT Server (1):
![mqtt](https://github.com/raomin/ESPAltherma/assets/89379538/1b711d13-42dd-4932-8777-ae44932b50ad)

**Comment by DieterB61** [comment]
I used now an ESP-32 Board, but although there GPIOs do not change there value when I try to set SG from espaltherma. Thus I'm quite sure that there is a problem with iobroker MQTT and espaltherma.

**Comment by DieterB61** [comment]
I found the problem. It was a wrong configuration of the MTTQ Server/Client of ioBroker that let me read the data but not write data. For users of ioBroker: take care to set the mask to publish correct (in my case it had to be mqtt.1.* as I used Adapter 1).

**Now my ioBroker works with a 4 relay board with ESP-12F (ESP8266) processor and the X85A extension to set energy levels :-) .**

For more information about my problem you can read (in German): https://forum.iobroker.net/topic/68950/gel%C3%B6st-senden-von-schaltzust%C3%A4nden-auf-ein-relaisboard

**Reply by DieterB61** [reply]
You can see the board here:
https://github.com/raomin/ESPAltherma/issues/17#issuecomment-1783834623

I ordered it at joom but it is sold out now:
https://www.joom.com/en/products/63447612f831cc01c88f0219

In fact I got no ESP32 but ESP-12 board which also works using the code available here. 
Search for "ESP32 relay board" to find one.

**Reply by lfrancke** [reply]
Thank you! I'm mostly new to hardware, this is useful.

**Comment by lfrancke** [comment]
Would you be willing to share information about the board you ordered?
